src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -472,6 +475,33 @@ object DefaultSource {
     )
   }
 
+  private[v1] def parseCogniteIds(paramName: String)(
+      parameters: Map[String, String]): Option[List[CogniteId]] = {
+
+    implicit val singleIdDecoder: Decoder[CogniteId] =
+      List[Decoder[CogniteId]](
+        Decoder.decodeLong
+          .validate(_.value.isNumber, "strict int decoder") // to distinguish 123 from "123"
+          .map(internalId => CogniteInternalId(internalId))
+          .widen,
+        Decoder.decodeString.map(externalId => CogniteExternalId(externalId)).widen
+      ).reduceLeft(_.or(_))
+    implicit val multipleIdsDecoder: Decoder[List[CogniteId]] =
+      List(
+        singleIdDecoder.map(List(_)),
+        Decoder.decodeArray[CogniteId].map(_.toList)
+      ).reduceLeft(_.or(_))
```

Comment:
neat 👍

Reply:
I don't think `.widen` is necessary since there is no covariance applicable here. 🤔